test/navigation/imu_preprocessing.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+
+#include <gtest/gtest.h>
+
+#include <core/logger.hpp>
+#include <core/types.hpp>
+#include <navigation/preprocess_imu.hpp>
+#include <utils/manual_time.hpp>
+
+namespace hyped::test {
+
+bool checkArrayEquality(core::ImuData &imu_data_a, core::ImuData &imu_data_b)
+{
+  if (imu_data_a.size() == imu_data_b.size()) {
+    for (std::size_t i; i < imu_data_a.size(); ++i) {
+      if (imu_data_a.at(i) == imu_data_b.at(i)) {
```

Comment:
Could just do 
```
if(imu_data_a.at(i) != imu_data_b.at(i)){
return false;
}
```
instead of the additional else statement?

Reply:
Also given that these are floats (of type `core::Float`), shouldn't we be using something along the lines of:
```
if(std::abs(imu_data_a.at(i) - imu_data_b.at(i)) < epsilon){
...
```
??

This feels like it would stop any potential failure of tests due to things like rounding errors at a very reasonable level of precision?